### ISSUE TITLE
feat(vulkan): execute independent Qwen3.8 decode lanes in one resident panel

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12743]
# feat(vulkan): execute independent Qwen3.8 decode lanes in one resident panel

```routing
lane: model
paths: ["internal/compute/qwen35_decode_batch_contract.go", "internal/compute/vulkan_qwen35_decode_batch.go", "internal/model/qwen35_batch_decode.go", "internal/model/qwen35_batch_decode_vulkan.go", "internal/compute/qwen35_decode_batch_test.go", "internal/model/qwen35_batch_decode_vulkan_test.go"]
expected_steps: 7
```

Parent: [fak #12389](https://github.com/anthony-chaudhary/fak/issues/12389). Campaign: [fak #8395](https://github.com/anthony-chaudhary/fak/issues/8395).

## Current state

At public base `d2e9acb066c06a54a30087ce307a7c0837c3b6e3`, `BatchSession.StepBatchActive` compacts active lanes, but the Qwen hybrid hook in `internal/model/qwen35_batch_decode.go` is installed only by the Darwin Metal implementation. A Vulkan build therefore falls through to one `Session.Step` call per lane and reports `LastStepSharedPanels()==0`.

The Vulkan Qwen sequence operation already sends token panels through shared resident projection kernels, but it owns one causal sequence, one KV store, and one set of recurrent states. Its token dimension cannot represent independent requests. The planner coordinator remains a separate agent-lane consumer after this capability lands.

Centrality: Core.

- P1 Context: preserved; each lane keeps its own token lineage, attention KV, convolution window, and Gated-DeltaNet recurrent state.
- P2 Net value: advanced; B=2..8 independent lanes share each resident projection weight stream.
- P3 Adaptation: preserved; B=1 and unsupported shapes keep the scalar route and report zero shared work.
- P4 Operations: advanced; typed execution counters distinguish shared projection panels from admission and serial fallback.

For native Qwen3.8 Vulkan serving / Problem: concurrent lanes repeat resident weight projections / Today: the only Qwen batch hook is Metal / Better because: one Vulkan operation advances independent lane state around shared B-row projections / Witness: independent software parity plus later repeated physical c4/c8 comparison.

## Core through-line

`BatchSession.StepBatchActive` -> exact Qwen/Vulkan capability check -> one typed B=2..8 backend decode-panel operation -> one B-row projection per resident weight with lane-owned attention and recurrence -> per-lane logits -> `LastStepSharedPanels` from observed backend execution.

Add a narrow compute contract for independent Qwen decode lanes and one Vulkan implementation using existing resident operations. Validate the complete lane set before mutation. After the first device submission or state mutation, any error is terminal for every participating session and must never replay through `Session.Step`. Report observed shared projection panels, device submissions, transfer bytes, and serial fallbacks. Model sets `lastStepSharedPanels` only after successful positive observed shared work with zero fallback.

## Gold-plating boundary

No new shader, native shim, scheduler, queue, prefill batching, speculative decoding, quantization format, B>8 support, default enablement, or comparison claim. Batching only the output head after serial full-model forwards does not satisfy this ticket. A wrapper around B scalar calls is not a panel.

## Done condition

- [ ] [SW-VERIFIED] B=2, B=4, B=8, and ragged active lanes use one mixed-layer Vulkan decode operation with independent KV and GDN state.
- [ ] [SW-VERIFIED] Every admitted resident projection consumes a B-row panel; the receipt reports positive shared panels, zero serial fallbacks, and zero activation D2H.
- [ ] [SW-VERIFIED] Per-lane accepted token, finite logits within predeclared tolerance, cache position and token lineage, full-attention KV, convolution state, and recurrent state match independent scalar references.
- [ ] [SW-VERIFIED] Full preflight rejects aliased state, mixed backend/model ownership, unsupported geometry, missing weights, B=1, and B>8 before mutation or shared-work claim.
- [ ] [SW-VERIFIED] An injected error after the first submission poisons participating sessions and performs no scalar replay.
- [ ] [SW-VERIFIED] Existing Metal batching and unsupported serial behavior remain unchanged.
- [ ] [HW-WITNESSED] A later source-bound N>=5 interleaved c4/c8 comparison preserves quality and retains the path only if aggregate accepted tok/s exceeds serialized execution by at least 5%.

## Witness

```text
rg -n "qwen35HybridQ4KBatchStep|stepBatchQwen35HybridQ4KMetal|LastStepSharedPanels|Session.Step" internal/model/qwen35_batch_decode.go internal/model/qwen35_batch_decode_metal.go internal/model/batch_step.go
```

This source witness shows one Darwin-only Qwen batch implementation and the serial fallback. It is not hardware evidence.

## Verifiable Witness

```text
go test ./internal/compute -run '^TestQwen35IndependentDecodeBatchContract' -count=1
go test -tags vulkan ./internal/model -run '^TestVulkanQwen35IndependentDecodeBatch' -count=1
```

Physical qualification requires explicit device selection and no skips. The witness must assert backend execution counters and `LastStepSharedPanels()>0`; admission size alone does not qualify.

## Likely files

- `internal/compute/qwen35_decode_batch_contract.go`
- `internal/compute/vulkan_qwen35_decode_batch.go`
- `internal/model/qwen35_batch_decode.go`
- `internal/model/qwen35_batch_decode_vulkan.go`
- Independently authored focused tests beside those packages.

## Lane

Public `internal/compute` and `internal/model`. Agent coordinator wiring follows after the model capability.

## Dependencies and dedupe

This is the bounded executable child of #12389 and does not duplicate closed Metal leaves #9074 or #9075. #12365 and #12098 retain true-concurrency scoring and hardware promotion. Open #12387 and #12388 remain promotion prerequisites; #12389 permits a bounded software slice while they proceed.

## Blast radius and affected lanes

Only optional Qwen hybrid Vulkan BatchSession execution is affected. CPU, CUDA, Metal, ordinary single-request decode, prefill, speculative decoding, and gateway behavior remain unchanged.

## Quarantined fallback mechanism

Unsupported envelopes decline before mutation and retain scalar execution with zero shared counters. An error after mutation is terminal and cannot trigger scalar replay. Default behavior stays unchanged.

## Acceptance gate

Independent contract and Vulkan model tests pass, focused compute/model suites and vet pass, and source review confirms that positive shared counters come only from B-row backend projections. Hardware promotion remains open until #12365/#12098 records the repeated matched gate.

## Closure binding

The resolving public commit references the child issue and #12389, includes only this compute/model capability plus independently authored tests, and carries the public model trailer. Documentation or serial fallback cannot close it.

## Working spine

1. Freeze typed request, result, and execution counters.
2. Validate B=2..8, exact backend/model identity, independent state ownership, and resident weights.
3. Execute mixed-layer Vulkan forward with shared B-row projections and lane-local state.
4. Fail closed after mutation and preserve truthful pre-submit fallback.
5. Bind the existing BatchSession Qwen hook on Vulkan.
6. Run independent parity, ownership, error, and counter witnesses.
7. Land the software slice; leave coordinator wiring and physical promotion separate.

## Work estimate

Estimate: 3 points.

## Overall completion contribution

Contribution: 3/8 points toward #12389; coordinator wiring and physical performance remain separate.

## Definition of done

The software leaf is complete when every SW-VERIFIED item is independently witnessed and the production path reports positive shared panels with zero serial fallback. Physical promotion remains open.

## Completion standard

development

## Parent context

#12389 owns the Vulkan ragged BatchSession objective and #8395 owns the serving baseline. This child supplies only the compute/model execution seam that #12389's latest audit found missing.

## Why this is next

The shared Vulkan projection kernels, independent session state objects, BatchSession compaction, and execution receipt already exist. The missing piece is one typed operation that binds those surfaces without adding a scheduler or shader.

## Work unit

leaf

## Expected steps

7

## Assumptions

- Existing Vulkan B-row matrix operations support every resident Qwen3.8 projection dtype and shape in B=2..8.
- Lane-local attention and GDN updates can consume shared projection rows without a new shader.
- The model hook can fail before mutation when the backend contract is absent.

## Confusion risks

- A causal token panel from one request is not a panel of independent requests.
- Positive cohort size without positive backend shared projection counters is not batch execution.
- A batch wrapper that calls Session.Step per lane is the existing fallback.

## Coordination

Coordinate with #12387 and #12388 around shared model/compute paths. The root task owns later agent coordinator wiring; this child owns no agent files.

## Trigger

Dispatch this child when the compute/model paths are free. Open physical promotion dependencies do not block its bounded software implementation.

## Batch policy

One existing-issue child for this dedupe key; update this issue instead of creating parallel Vulkan lane-panel tickets.

## Generation

G1

## Work scale

S1

## Required model tier

tier/T2-required

## Optimal model tier

tier/T1-optimal
